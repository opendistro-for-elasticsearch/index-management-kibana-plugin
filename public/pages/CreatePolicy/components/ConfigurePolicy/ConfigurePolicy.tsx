/*
 * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React, { ChangeEvent } from "react";
import { EuiSpacer, EuiFormRow, EuiFieldText, EuiText } from "@elastic/eui";
import { ContentPanel } from "../../../../components/ContentPanel";

interface ConfigurePolicyProps {
  policyId: string;
  policyIdError: string;
  isEdit: boolean;
  onChange: (value: ChangeEvent<HTMLInputElement>) => void;
}

const ConfigurePolicy = ({ isEdit, policyId, policyIdError, onChange }: ConfigurePolicyProps) => (
  <ContentPanel bodyStyles={{ padding: "initial" }} title="Name policy" titleSize="s">
    <div style={{ paddingLeft: "10px" }}>
      <EuiText size="xs">
        <p>Policies let you automatically perform administrative operations on indices.</p>
      </EuiText>
      <EuiSpacer size="s" />
      <EuiFormRow
        label="Policy ID"
        helpText="Specify a unique ID that is easy to recognize and remember."
        isInvalid={!!policyIdError}
        error={policyIdError}
      >
        <EuiFieldText isInvalid={!!policyIdError} placeholder="hot_cold_workflow" readOnly={isEdit} value={policyId} onChange={onChange} />
      </EuiFormRow>
    </div>
  </ContentPanel>
);

// @ts-ignore
export default ConfigurePolicy;
