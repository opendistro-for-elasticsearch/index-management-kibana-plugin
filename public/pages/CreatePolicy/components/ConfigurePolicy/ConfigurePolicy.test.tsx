/*
 * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from "react";
import { render } from "@testing-library/react";
import ConfigurePolicy from "./ConfigurePolicy";

describe("<ConfigurePolicy /> spec", () => {
  it("renders the component", () => {
    const { container } = render(<ConfigurePolicy policyId="some_id" policyIdError="" isEdit={false} onChange={() => {}} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
