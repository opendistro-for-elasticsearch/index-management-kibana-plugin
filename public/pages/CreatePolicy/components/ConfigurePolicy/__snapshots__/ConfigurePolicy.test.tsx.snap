// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfigurePolicy /> spec renders the component 1`] = `
<div
  class="euiPanel euiPanel--paddingMedium"
  style="padding-left: 0px; padding-right: 0px;"
>
  <div
    class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
    style="padding: 0px 10px;"
  >
    <div
      class="euiFlexItem"
    >
      <h3
        class="euiTitle euiTitle--small"
      >
        Name policy
      </h3>
    </div>
  </div>
  <hr
    class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginXSmall"
  />
  <div
    style="padding: initial;"
  >
    <div
      style="padding-left: 10px;"
    >
      <div
        class="euiText euiText--extraSmall"
      >
        <p>
          Policies let you automatically perform administrative operations on indices.
        </p>
      </div>
      <div
        class="euiSpacer euiSpacer--s"
      />
      <div
        class="euiFormRow"
        id="some_make_id-row"
      >
        <label
          aria-invalid="false"
          class="euiFormLabel"
          for="some_make_id"
        >
          Policy ID
        </label>
        <div
          class="euiFormControlLayout"
        >
          <div
            class="euiFormControlLayout__childrenWrapper"
          >
            <input
              aria-describedby="some_make_id-help"
              class="euiFieldText"
              id="some_make_id"
              placeholder="hot_cold_workflow"
              type="text"
              value="some_id"
            />
          </div>
        </div>
        <div
          class="euiFormHelpText euiFormRow__text"
          id="some_make_id-help"
        >
          Specify a unique ID that is easy to recognize and remember.
        </div>
      </div>
    </div>
  </div>
</div>
`;
